### ISSUE TITLE
kover should able to exclude interface default method

### ISSUE THREAD

**Issue by hanrw** [Kotlin/kotlinx-kover#623]
It's nice to have a filter for excluding the interface which has a default implementation
E.g
```kotlin
package org.koin.core.component

import org.koin.core.Koin
import org.koin.core.parameter.ParametersDefinition
import org.koin.core.qualifier.Qualifier
import org.koin.mp.KoinPlatformTools

/**
 * KoinComponent interface marker to bring Koin extensions features
 *
 * @author Arnaud Giuliani
 */
interface KoinComponent {

    /**
     * Get the associated Koin instance
     */
    fun getKoin(): Koin = KoinPlatformTools.defaultContext().get()
}


data class AnthropicConfig(
    val apiKey: () -> String = { "CONFIG_API_KEY" },
    val baseUrl: () -> String = { Anthropic.BASE_URL },
    val anthropicVersion: () -> String = { Anthropic.ANTHROPIC_VERSION },
) : KoinComponent
```
[Check here](https://app.codecov.io/gh/tddworks/openai-kotlin/blob/main/anthropic-client%2Fanthropic-client-core%2Fsrc%2FcommonMain%2Fkotlin%2Fcom%2Ftddworks%2Fanthropic%2Fapi%2FAnthropicConfig.kt)
<img width="1313" alt="image" src="https://github.com/Kotlin/kotlinx-kover/assets/1201118/7ff12e9e-089e-41b5-a930-3631ba81d72f">

the cover is not covered unless added a ut like below
```kotlin
class AnthropicConfigTest : AutoCloseKoinTest() {
    
    @Test
    fun `only fix the coverage by this case`() {
        startKoin {
            val r = AnthropicConfig().getKoin()
            assertNotNull(r)
        }
    }
}
```

**Comment by shanshin** [comment]
We will consider adding similar filters in the future.

But could you describe your case in more detail, for what purpose do you have methods with default implementation in interfaces, but it is acceptable for you not to use them?

**Comment by hanrw** [comment]
I discovered the issue after introducing Koin(https://github.com/InsertKoinIO/koin) as the dependency injection framework in my project. and found the interface KoinComponent has a default implementation.

**Comment by shanshin** [comment]
Do I understand correctly that the `KoinComponent` interface does not belong to your project and you did not write it?
In this case, you should exclude it from the reports.

The collection of coverage is primarily aimed at finding unused code that the project developer wrote. 
If any external dependencies are appeared in the report, they should be completely exclude.

**Comment by hanrw** [comment]
yes. you are right.  but i have my own code which extends external code like `KoinComponent` for this case is there better solution from your point of view?
let's say for the class AnthropicConfig I had covered call cases but the kover report that there's something that not cover yet.

**Comment by shanshin** [comment]
> let's say for the class AnthropicConfig I had covered call cases but the kover report that there's something that not cover yet.

could you provide an XML report, at least with class `AnthropicConfig` coverage?

The report you provided is generated using codecov, we do not control it directly, it may have its own rules for calculating coverage

**Comment by shanshin** [comment]
It's bug in Kover reporter.

Reproducer
```kortin

interface Parent {
    fun getFoo(): String = "foo"
}


data class Child(
    val s: () -> String = { "Text" },
) : Parent

```

report 
```
<class name="org/jetbrains/kover/android/test/Child" sourcefilename="Extensions.kt">
<method name="&lt;init&gt;" desc="(Lkotlin/jvm/functions/Function0;)V">
<counter type="INSTRUCTION" missed="8" covered="0"/>
<counter type="BRANCH" missed="0" covered="0"/>
<counter type="LINE" missed="1" covered="0"/>
</method>
<method name="getFoo" desc="()Ljava/lang/String;">
<counter type="INSTRUCTION" missed="2" covered="0"/>
<counter type="BRANCH" missed="0" covered="0"/>
<counter type="LINE" missed="1" covered="0"/>
</method>
<counter type="INSTRUCTION" missed="10" covered="0"/>
<counter type="BRANCH" missed="0" covered="0"/>
<counter type="LINE" missed="2" covered="0"/>
<counter type="METHOD" missed="2" covered="0"/>
</class>
```

**Comment by hanrw** [comment]
thx.

**Comment by kszorin** [comment]
@shanshin , hi! Could you tell about approximate time for bug fix?

**Comment by shanshin** [comment]
Closed due to the end of IntelliJ Agent development (#720) and the introduction of a new strategy for measuring of Kotlin code coverage (#746).

It is recommended to use JaCoCo, for this purpose it is enough to specify
```kotlin
kover {
    useJacoco("0.8.13")
}
```
in each project.

Based on our internal testing, this issue does not appear to be reproducible in JaCoCo.
